apps/pipelines/tests/test_models.py

```diff
@@ -9,3 +12,37 @@ def test_archive_pipeline_archives_nodes_as_well():
     assert pipeline.node_set.count() > 0
     pipeline.archive()
     assert pipeline.node_set.count() == 0
+
+
+@pytest.mark.django_db()
+class TestNode:
+    @pytest.mark.parametrize("versioned_assistant_linked", [True, False])
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_versioning_assistant_node(self, versioned_assistant_linked):
+        """
+        Versioning an assistant node should version the assistant as well, but only when the linked assistant is not
+        already a version
+        """
```

Comment:
> only when the linked assistant is not already a version
when would this case happen? Wouldn't this be versioned at the same time as the experiment when the assistant versioning happens as well?

Reply:
This can happen when I for instance have an assistant version that I'd like to call in the pipeline. Not 100% how often this would be the case, but it is a real possibility